wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift

```diff
@@ -90,40 +122,50 @@ class MLSEventProcessor: MLSEventProcessing {
         }
 
         mlsService.registerPendingJoin(groupID)
-        Logging.mls.info("MLS event processor added group (\(groupID)) to be joined")
+        WireLogger.mls.info("MLS event processor added group (\(groupID.safeForLoggingDescription)) to be joined")
     }
 
     // MARK: - Process welcome message
 
-    func process(welcomeMessage: String, in context: NSManagedObjectContext) {
-        Logging.mls.info("MLS event processor is processing welcome message")
+    public func process(
+        welcomeMessage: String,
+        conversationID: QualifiedID,
+        in context: NSManagedObjectContext
+    ) {
+        WireLogger.mls.info("MLS event processor is processing welcome message")
 
         guard let mlsService = context.mlsService else {
             return logWarn(aborting: .processingWelcome, withReason: .missingMLSService)
         }
 
         do {
             let groupID = try mlsService.processWelcomeMessage(welcomeMessage: welcomeMessage)
-
-            guard let conversation = ZMConversation.fetch(with: groupID, in: context) else {
-                return logWarn(aborting: .processingWelcome, withReason: .other(reason: "conversation does not exist in db"))
+            mlsService.uploadKeyPackagesIfNeeded()
+
+            if let conversation = ZMConversation.fetch(with: conversationID, in: context) {
+                conversation.mlsGroupID = groupID
+                conversation.mlsStatus = .ready
+                WireLogger.mls.info("MLS event processor set mlsStatus to ready for group \(groupID.safeForLoggingDescription)")
+                context.saveOrRollback()
+            } else {
+                // Conversation doesn't exist locally yet, so fetch it from the backend.
+                // It'll be marked as ready when it's synced locally.
+                conversationService.syncConversation(qualifiedID: conversationID) {
+                    // no op
+                }
```

Comment:
```suggestion
                conversationService.syncConversation(qualifiedID: conversationID)
```
I think the completion has default {}

Reply:
Not anymore since the type is now `ConversationServiceInterface` rather than the concrete implementation that has the default.

Reply:
ah ok